### ISSUE TITLE
Lösung suchen für das Zurücksetzen von Steuerung UND den Toren

### ISSUE THREAD

**Issue by uefvanscheld** [uefvanscheld/Rudis_Tor#16]

**Comment by uefvanscheld** [comment]
ja, langes Drücken reicht (extra-lang war nicht so wörtlich gemeint - einfach nur deutlich länger als normal, damit es nicht unabsichtlich passiert ). Die Taste der der Fernbedienung reicht sicher auch, wenn wir uns auf eine Richtung beschränken können (oder wir kodieren die Richtung über die Dauer des Drucks: 
- lang: Öffnen bis zum Anschlag
- extra-lang: Schließen bis zum Anschlag

**Comment by yuoheim** [comment]
Hier haben wir ein Missverständnis. Ich hatte das so verstanden, dass du beide Tasten gleichzeitig lange drücken wolltest und nicht dass du damit die Richtung wählen wolltest.
Extra langes Drücken halten ich schon für gut, es sollten schon ein paar Sekunden sein, damit ist eine Fehlbedienung ziemlich unwahrscheinlich.
Das Öffnen ist wohl der unkritischere Fall fürs Tor, hierbei kann es sich eigentlich nicht verklemmen. Eigentlich sollte das auch reichen um wieder in einen definierten Zustand zu kommen, wenn die Tore dann bis zum Anschlag gefahren sind.
Wenn man es jetzt auch noch verhindern möchten das etwas be- oder jemand geschädigt wird, kann man auch verlangen dass die Taste die ganze Zeit gedrückt bleibt und wenn man sie los lässt, stoppen die Tore.
Ist nur die Frage wie weit wir es treiben wollen. Im ersten Step sollte es reichen wieder einen definierten Zustand zu erreichen damit man weiter testen bzw. das Tor wieder bedienen kann.

**Comment by yuoheim** [comment]
Stoppen durch erneutes Drücken der Taste reicht völlig!

Im Moment sehe ich auch nur die Möglichkeit dass die Tore beim Loslassen der Taste losfahren, wenn wir die Dauer des Drückens auswerten.

Ich frag Rudi mal, ob die Fernbedienung vielleicht noch einen zweiten Kanal hat. Diesen könnte man dann an den "Hand" Eingang oder noch freie DI anschließen und für solche Funktionen nutzen.

**Comment by yuoheim** [comment]
Ich habe gerade mit Rudi gesprochen. Also der manuelle Taster "Hand" steht auch später zur Verfügung wenn die Steuerung eingebaut ist. Rudi hat dafür einen Taster im Haus, der über eine Mehraderleitung mit der Steuerung in einem der Pfosten am Tor verbunden ist.
Ferner hat er sich auch eine andere Fernsteuerung besorgen wo jetzt noch ein weiterer Kanal zur Verfügung stehen würde. Diesen könnten man z. B. mit A7 verbinden.
Dann haben wir bis auf Tx und Rx alle E/As des Nanos genutzt.
Eine Idee für die Nutzung des zusätzlichen Kanals hat er auch schon. Er meint damit nur das rechte Tor zusteuern, wenn man mal nur so dadurch muss. Ich habe ihn gebeten uns erstmal die anderen Funktionen realisieren zu lassen.
